src/main/client_context.cpp

```diff
@@ -491,6 +522,21 @@ void ClientContext::commitUDFTrx(bool isAutoCommitTrx) {
     }
 }
 
+bool ClientContext::canExecuteWriteQuery() {
+    if (dbConfig.readOnly) {
+        return false;
+    }
+    // Note: we can only attach a remote kuzu database in read-only mode and only one
+    // remote kuzu database can be attached. If there is one remote
+    auto dbManager = database->databaseManager.get();
+    for (auto& attachedDB : dbManager->getAttachedDatabases()) {
+        if (attachedDB->getDBType() == common::ATTACHED_KUZU_DB_TYPE) {
```

Comment:
hmm should we do the check and throw exception during attaching? so users have to specify read only for attached remote kuzu db. that's more user friendly.

Reply:
I don't think so, we can mention this in the doc so users know that they can only execute read-only queries inside attached database.